azurerm/resource_arm_automation_credential.go

```diff
@@ -41,10 +41,23 @@ func resourceArmAutomationCredential() *schema.Resource {
 
 			"resource_group_name": azure.SchemaResourceGroupName(),
 
+			//this is AutomationAccountName in the SDK
 			"account_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				Deprecated:    "account_name has been renamed to automation_account_name for clarity and to match the azure API",
+				ConflictsWith: []string{"automation_account_name"},
+				ValidateFunc:  azure.ValidateAutomationAccountName(),
+			},
+
+			"automation_account_name": {
 				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Optional: true, //todo change to required once account_name has been removed
+				Computed: true,
+				//ForceNew:      true, //todo this needs to come back once account_name has been removed
```

Comment:
As we are removing this, it might be worth catching the error on update and checking if its for renaming this property and adding a `the name property needs a manual destroy/re-creation to update for until 2.0` or something to that effect?

Reply:
I was about to implement this and then realised that this should just work when the `ForceNew` property is set to `true` on both. I've tested it and it works as one expects. Even when changing between `account_name` and `automation_account_name`

Reply:
The use case i would be concerned with there is migrating to the new property. If i am on the old one and then switch to the new property would that cause a recreation?

Reply:
From what I tested, it doesn't. This is probably due to the fact that both are computed. This means that if you don't specify any value, it uses the current value. Switching between account_name and automation_account_name and back doesn't change anything (unless you change the associated value).

Reply:
perfect! 👌